certbot/certbot/_internal/main.py

```diff
@@ -1652,6 +1653,76 @@ def make_or_verify_needed_dirs(config: configuration.NamespaceConfig) -> None:
         util.make_or_verify_dir(hook_dir, strict=config.strict_permissions)
 
 
+def reconfigure(config: configuration.NamespaceConfig,
+          plugins: plugins_disco.PluginsRegistry) -> None:
+    """Allow the user to set new configuration options for an existing certificate without
+       forcing renewal. This can be used for things like authenticator, installer, and hooks,
+       but not for the domains on the cert, since those are only saved in the cert.
+
+    :param config: Configuration object
+    :type config: configuration.NamespaceConfig
+
+    :param plugins: List of plugins
+    :type plugins: plugins_disco.PluginsRegistry
+
+    :raises errors.Error: if the dry run fails
+    :raises errors.ConfigurationError: if certificate could not be loaded
+
+    """
+
+    if config.domains:
+        raise errors.ConfigurationError("You have specified domains, but this function cannot "
+            "be used to modify the domains in a certificate. If you would like to do so, you "
+            "will need to renew the certificate instead of reconfiguring, which can be done "
+            "with the flag --force-renewal. Otherwise, remove the domains from the command "
```

Comment:
Is `--force-renewal` actually necessary? I would recommend to mention `--force-renewal` as little as possible, as users on the Community often don't understand its meaning, purpose and potential rate limit hazard.

That said, the `renew` subcommand also doesn't work with `-d` options ("Currently, the renew verb is capable of either renewing all installed certificates that are due to be renewed or renewing a single certificate specified by its name. If you would like to renew specific certificates by their domains, use the certonly command instead. The renew verb may provide other options for selecting certificates to renew in the future.")

Users are required to use the `certonly` subcommand and with that, Certbot will ask to *update* the cert, even without `--force-renewal`.

Reply:
I mean, kind of. `--force-renewal` is just the flag to bypass the "are you sure you want to do this" question, so those are essentially the same thing, just one you tell it interactively and one you tell it on the command line. We can certainly say "which can be done by running `certbot certonly` instead of `certbot reconfigure`" and let that all happen interactively though, if we want to minimize references to `--force-renewal`.

Reply:
I agree with Osiris here in terms of this phrase:

>you will need to renew the certificate instead

The `renew` verb is not applicable here and it's not really a renew operation. It's going to be `certonly` or `run` with `--cert-name` set and `--force-renewal`. Quite a mouthful.

My proposal: create a new certbot.org short link and link to https://eff-certbot.readthedocs.io/en/stable/using.html#changing-a-certificate-s-domains from this error message. And make sure that section doesn't get wiped out by Alexis' work in #9439.

Reply:
I still think we should change the language here.

To say "you will need to renew the certificate instead" while our `renew` verb is not actually capable of changing domains on the certificate, will end up being a wild goose chase for the user.